zerver/lib/message.py

```diff
@@ -1110,6 +1110,7 @@ def extract_unread_data_from_um_rows(
         return raw_unread_messages
 
     muted_stream_ids = get_muted_stream_ids(user_profile)
+    raw_unread_messages["muted_stream_ids"] = muted_stream_ids
```

Comment:
This bug fix would have been much better to submit an independent PR for, since it's fixing a regression that's in `main`; the bug in question went out to production last night because I hadn't had a chance to re-review this.